src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeServiceClient.java

```diff
@@ -0,0 +1,226 @@
+
+  // Request builder for building streaming API request
+  private final RequestBuilder requestBuilder;
+
+  /**
+   * Default constructor
+   *
+   * @param httpClient the HTTP client used for making requests
+   * @param requestBuilder the request builder for building streaming API requests
+   */
+  SnowflakeServiceClient(CloseableHttpClient httpClient, RequestBuilder requestBuilder) {
+    this.httpClient = httpClient;
+    this.requestBuilder = requestBuilder;
+  }
+
+  /**
+   * Configures the client given a {@link ConfigureRequest}.
+   *
+   * @param request the client configuration request
+   * @return the response from the configuration request
+   */
+  ConfigureResponse clientConfigure(ConfigureRequest request) {
+    ConfigureResponse response =
+        executeApiRequestWithRetries(
+            request,
+            ConfigureResponse.class,
+            CLIENT_CONFIGURE_ENDPOINT,
+            "client configure",
+            STREAMING_CLIENT_CONFIGURE,
+            ErrorCode.CONFIGURE_FAILURE);
+    if (response.getStatusCode() != RESPONSE_SUCCESS) {
+      logger.logDebug("Client configure request failed, message={}", response.getMessage());
+      throw new SFException(ErrorCode.CONFIGURE_FAILURE, response.getMessage());
+    }
+    return response;
+  }
+
+  /**
+   * Configures the channel given a {@link ConfigureRequest}.
+   *
+   * @param request the channel configuration request
+   * @return the response from the configuration request
+   */
+  ConfigureResponse channelConfigure(ConfigureRequest request) {
+    ConfigureResponse response =
+        executeApiRequestWithRetries(
+            request,
+            ConfigureResponse.class,
+            CHANNEL_CONFIGURE_ENDPOINT,
+            "channel configure",
+            STREAMING_CHANNEL_CONFIGURE,
+            ErrorCode.CONFIGURE_FAILURE);
+
+    if (response.getStatusCode() != RESPONSE_SUCCESS) {
+      logger.logDebug(
+          "Channel configure request failed, table={}, message={}",
+          request.getFullyQualifiedTableName(),
+          response.getMessage());
+      throw new SFException(ErrorCode.CONFIGURE_FAILURE, response.getMessage());
+    }
+    return response;
+  }
+
+  /**
+   * Opens a channel given a {@link OpenChannelRequestInternal}.
+   *
+   * @param request the open channel request
+   * @return the response from the open channel request
+   */
+  OpenChannelResponse openChannel(OpenChannelRequestInternal request) {
+    OpenChannelResponse response =
+        executeApiRequestWithRetries(
+            request,
+            OpenChannelResponse.class,
+            OPEN_CHANNEL_ENDPOINT,
+            "open channel",
+            STREAMING_OPEN_CHANNEL,
+            ErrorCode.OPEN_CHANNEL_FAILURE);
+
+    if (response.getStatusCode() != RESPONSE_SUCCESS) {
+      logger.logDebug(
+          "Open channel request failed, table={}, message={}",
+          request.getFullyQualifiedTableName(),
+          response.getMessage());
```

Comment:
old code logs more info:
```
        logger.logDebug(
            "Open channel request failed, channel={}, table={}, client={}, message={}",
            request.getChannelName(),
            request.getFullyQualifiedTableName(),
            getName(),
            response.getMessage());
```

suggest adding a method to the StreamingIngestRequest interface, called "getStringForLogging(), forcing every request to decide which of its' properties it wants to write out to logs.

Reply:
Added `getStringForLogging()` for every request.